akka-http-core/src/test/scala/akka/http/impl/engine/client/ConnectionPoolSpec.scala

```diff
@@ -566,4 +591,45 @@ class ConnectionPoolSpec extends AkkaSpec("""
   }
 
   object NoErrorComplete extends SingletonException
+
+  abstract class ClientTransportTestSetup {
+    def issueRequest(request: HttpRequest, settings: ConnectionPoolSettings): Future[HttpResponse]
+
+    class CustomTransport extends ClientTransport {
+      val in = ByteStringSinkProbe()
+      val out = TestPublisher.probe[ByteString]()
+      val promise = Promise[(String, Int, ClientConnectionSettings)]()
+
+      def connectTo(host: String, port: Int, settings: ClientConnectionSettings)(implicit system: ActorSystem): Flow[ByteString, ByteString, Future[OutgoingConnection]] = {
+        promise.success((host, port, settings))
+        Flow.fromSinkAndSource(in.sink, Source.fromPublisher(out)).mapMaterializedValue(_ ⇒ Promise().future)
+      }
+    }
+
+    import system.dispatcher
+
+    val transport = new CustomTransport
+    val poolSettings =
+      ConnectionPoolSettings(system)
+        .withTransport(transport)
+        .withConnectionSettings(ClientConnectionSettings(system).withIdleTimeout(23.hours))
```

Comment:
extract the `23.hours` as value? a bit easier to see where it's set / expected then?